src/vm/callables.rs

```diff
@@ -97,12 +110,31 @@ impl DefinedFunction {
     }
 
     pub fn get_identifier(&self) -> FunctionIdentifier {
-        let (body, arguments) = match self {
-            DefinedFunction::Private(f) => (f.body.clone(), f.arguments.clone()),
-            DefinedFunction::Public(f) => (f.body.clone(), f.arguments.clone())
-        };
-        return FunctionIdentifier {
-            body: body,
-            arguments: arguments }
+        match self {
+            DefinedFunction::Public(f) => f.identifier.clone(),
+            DefinedFunction::Private(f) => f.identifier.clone()
+        }
+    }
+}
+
+impl CallableType {
+    pub fn get_identifier(&self) -> FunctionIdentifier {
+        match self {
+            CallableType::UserFunction(f) => f.get_identifier(),
+            CallableType::NativeFunction(s, _) => FunctionIdentifier::new_native_function(s),
```

Comment:
Not a bug, but IIRC the convention for instantiating a struct in Rust from another struct is to use a name like `from_native_function` instead of `new_native_function`.

Reply:
Ah, okay, that makes sense. Will standardize my instantiators/constructors.